crates/matrix-sdk-base/src/sliding_sync.rs

```diff
@@ -124,11 +124,14 @@ impl BaseClient {
     ///   sync.
     /// * `previous_events_provider` - Timeline events prior to the current
     ///   sync.
+    /// * `ignore_verification_requests` - Whether verification requests found
+    ///   in the response should be ignored.
     #[instrument(skip_all, level = "trace")]
     pub async fn process_sliding_sync<PEP: PreviousEventsProvider>(
         &self,
         response: &http::Response,
         previous_events_provider: &PEP,
+        ignore_verification_requests: bool,
```

Comment:
Can't we get this information from the `BaseClient` instead of passing an argument everywhere?

Reply:
Ah, here is the reasoning about the parameter, I'm not sure this is related to sliding sync, it has more to do with the notification client.

While our notification client exclusively uses sliding sync, one could imagine one that uses classic sync.

Reply:
Which one? The `BaseClient`? We already do so:

https://github.com/matrix-org/matrix-rust-sdk/blob/19df945155cefe80dada6c18e341cd5d1fe5727d/crates/matrix-sdk/src/client/mod.rs#L2423-L2426

We could set the flag there, then the public API of the `SlidingSyncBuilder` wouldn't need to be changed.

Reply:
Oh ok, that makes sense! Let me try that instead then.

Reply:
Ok, so I think the latest changes should handle this the way you and @Hywan wanted it to. Could you check?